node/src/test/scala/com/wavesplatform/state/diffs/SetScriptTransactionDiffTest.scala

```diff
@@ -62,12 +53,97 @@ class SetScriptTransactionDiffTest extends PropSpec with WithDomain {
+      d.appendAndCatchError(TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeContract(StdLibVersion.V5, 32769), version = TxVersion.V2))
+        .toString should include(
+        "32769 bytes > 32768 bytes"
+      )
+
+      d.appendAndAssertSucceed(TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeExpr(StdLibVersion.V5, 8192), version = TxVersion.V2))
+      d.appendAndCatchError(TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeExpr(StdLibVersion.V5, 8193), version = TxVersion.V2))
+        .toString should include(
+        "Script is too large: 8193 bytes > 8192 bytes"
+      )
+    }
+  }
+
+  property("limit 160kb/8kb after V6") {
+    withDomain(DomainPresets.RideV6) { d =>
+      d.helpers.creditWavesToDefaultSigner()
+
+      val setScript160kb = TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeContract(StdLibVersion.V6, 160 * 1024), version = TxVersion.V2)
+      d.commonApi.calculateWavesFee(setScript160kb) shouldBe 0.16.waves
+      d.appendAndAssertSucceed(setScript160kb)
+
+      d.appendAndCatchError(TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeContract(StdLibVersion.V6, 160 * 1024 + 1), version = TxVersion.V2))
+        .toString should include(
+        "Script is too large: 163841 bytes > 163840 bytes"
+      )
+
+      d.appendAndAssertSucceed(TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeExpr(StdLibVersion.V6, 8 * 1024), version = TxVersion.V2))
+      d.appendAndCatchError(TxHelpers.setScript(TxHelpers.defaultSigner, exactSizeExpr(StdLibVersion.V6, 8 * 1024 + 1), version = TxVersion.V2))
+        .toString should include(
+        "Script is too large: 8193 bytes > 8192 bytes"
+      )
+    }
+
+    def byteVectorsList(size: Int): String =
+      (1 to size).map(_ => s"base64'${ByteStr(new Array[Byte](1000)).base64Raw}'").mkString("[", ", ", "]")
+
+    intercept[RuntimeException](TxHelpers.exprScript(StdLibVersion.V6)(s"""
+                                                                          |strict a = ${byteVectorsList(9)}
+                                                                          |true
+                                                                          |""".stripMargin)).toString should include(
+      "Script is too large: 9140 bytes > 8192 bytes"
+    )
+  }
+
+  property("lowered fee after V6") {
+    val script = TxHelpers.scriptV6(s"""
+                                       |@Callable(i)
+                                       |func test() = {
+                                       |  []
+                                       |}
+                                       |""".stripMargin)
+
+    withDomain(DomainPresets.RideV6) { d =>
+      val setScriptTransaction = TxHelpers.setScript(TxHelpers.defaultSigner, script, version = TxVersion.V2)
+      d.commonApi.calculateWavesFee(setScriptTransaction) shouldBe 0.001.waves
+    }
+  }
 
```

Comment:
Add test for dynamic fee calculation:
- 1024 bytes should be 0.001 waves
- 1025 bytes should be 0.002 waves
- 32/160kb before/after V6 (max) - 0.032 and 0.16 waves respectively